### PR TITLE
Enhance Config Serialization

### LINKED ISSUE #1279: Expected a list, but got a scalar value
**Describe the bug**

**To Reproduce**
Steps to reproduce the bug:
1. 使用一个symbols键值对中值为scalar对象的(补充)词表，如https://github.com/wklchris/Rime-latex-symbols ，具体例如：

```
punctuator:
  symbols:
    '/angle': ∠
    # 改为'/angle': [∠]则正常
```

2. 使用Trime，正常等待自动部署
3. 连续弹窗报错，无法正常使用，甚至很难关闭

**Expected behavior**
能够识别scalar对象而不一定要求list，从而与一些之前的词表兼容，不发生部署连环报错。

**Log**
[com.osfans.trime-2024-03-04T18_40_32Z.txt](https://github.com/osfans/trime/files/14498441/com.osfans.trime-2024-03-04T18_40_32Z.txt)

**Smartphone (please complete the following information):**
 - Device: Redmi K30i 5G
 - OS: Android 13
 - Theme: Tongwenfeng
 - Version: 3.2.17(and nighly)
 - Build info:
 
 ```
Builder: Nightly CI
Git Repo: https://github.com/osfans/trime
Build Git Hash: 913420859d690a71a6003f857ae88d237ca20109
Build Time: Mar 5, 2024 00:48:51
 ```

### PR DESCRIPTION
**Pull request by WhiredPlanck**
#### Issue tracker
Fixes will automatically close the related issues

Fixes #1279 

#### Feature
Describe features of this pull request

Currently trime depends on librime's APIs to read configs that complied with librime DSL for yaml, especially the theme configs. But bridging C++ API to Java/Kotlin can be really painful, and we get a lot of unchecked-cast warnings and type judging.

I've implemented Config class before, which the same thing in librime rewritten in Kotlin. I decide to utilize it to make devs' life easier.

#### Code of conduct
- [x] [CONTRIBUTING](CONTRIBUTING.md)

#### Style lint
- [x] `make sytle-lint`

#### Build pass
- [x] `make debug`

#### Manually test
- [x] Done

#### Code Review
1. No wildcards import
2. Manual build and test pass
3. GitHub Action CI pass
4. At least one contributor review and approve
5. Merged clean without conflicts
6. PR will be merged by rebase upstream base

#### Daily build
Login and download artifact at https://github.com/osfans/trime/actions

### CODE REVIEW

**Comment by nopdan** [comment]
你是不是把 rime 配置文件想得太简单了？这些都实现了吗 https://docs.nopdan.com/rime/Configuration.html

**Comment by WhiredPlanck** [comment]
> 你是不是把 rime 配置文件想得太简单了？这些都实现了吗 https://docs.nopdan.com/rime/Configuration.html

不需要实现呀，librime 的 API 和我自造的 Config 类读取的都是**部署编译过后的**的文件（也就是 build 目录里的对应文件），编译过后的配置是标准的 yaml 文件，直接读就是了。

File: app/src/main/java/com/osfans/trime/data/theme/FontManager.kt [hunk 1]
```diff
@@ -63,7 +65,14 @@ object FontManager {
     }
 
     private fun getTypefaceOrDefault(key: String): Typeface {
-        val fonts = ThemeManager.activeTheme.style.getObject(key)
+        val fonts =
+            ThemeManager.activeTheme.style.getItem(key).run {
```
Comment (goofyz) [inline app/src/main/java/com/osfans/trime/data/theme/FontManager.kt:69]: Many classes use `ThemeManager.activeTheme` to get the current `theme`.  Can we use `kotlin-inject` to inject `theme` in these places?

**Review by goofyz — APPROVED** [review]

File: app/src/main/java/com/osfans/trime/data/theme/FontManager.kt (hunk 1 above)
Comment (WhiredPlanck) [reply app/src/main/java/com/osfans/trime/data/theme/FontManager.kt:69]: Need to verify.

**Pull request merged**